lib/lotus/model/migrator/mysql_adapter.rb

```diff
@@ -9,13 +9,13 @@ class MySQLAdapter < Adapter
         # @since 0.4.0
         # @api private
         def create
-          new_connection.run %(CREATE DATABASE #{ database };)
+          new_connection(global: true).run %(CREATE DATABASE #{ database };)
```

Comment:
Not really related with permission. When creating a database we don't have it yet (I felt mr. obvious saying that), so we can't connect directly to the database. We're handling it by always using a _global_ connection: https://github.com/lotus/model/blob/4d48f4b24ee305e966ebc6d7b29ce556c9faeec3/lib/lotus/model/migrator/adapter.rb#L103-L108 as we can see in this code no database name is specified.

Summarising, we just need _global_ connections for creating and dropping databases :)